Library/Homebrew/download_strategy.rb

```diff
@@ -389,6 +389,23 @@ def _fetch
   end
 end
 
+# Will convert "https://downloads.sourceforge.net/project/..." URLs to mirrorservice.org
+# In case environment variable HOMEBREW_NO_INSECURE_REDIRECT is set, so the Sourceforge is not ultimately broken
+class CurlSourceForgeSecureMirrorDownloadStrategy < CurlDownloadStrategy
+  def fetch
+    unless ENV["HOMEBREW_NO_INSECURE_REDIRECT"].nil?
```

Comment:
You only append mirrors, so it won't alter the normal downloading processing. In the mean time, it will help to mitigate the downloading problems, say sourceforge is down.